tensornetwork/backends/shell/shell_backend.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import operator
+from tensornetwork.backends import base_backend
+from typing import Optional, Sequence, Tuple, List, Any
+
+
+class ShellTensor:
+  def __init__(self, shape: Tuple):
+    self.shape = shape
+
+  def reshape(self, new_shape: Tuple):
+    self.shape = new_shape
+    return self
+
+
+Tensor = "ShellTensor"
+
+
+class ShellBackend(base_backend.BaseBackend):
+  """See base_backend.BaseBackend for documentation."""
+
+  def __init__(self):
+    super(ShellBackend, self).__init__()
+    self.name = "shell"
+
+  def tensordot(self, a: Tensor, b: Tensor,
+                axes: Sequence[Sequence[int]]) -> Tensor:
+    # Does not work when axis < 0
+    gen_a = (x for i, x in enumerate(a.shape) if i not in axes[0])
+    gen_b = (x for i, x in enumerate(b.shape) if i not in axes[1])
+    return ShellTensor(self._concat_generators(gen_a, gen_b))
+
+  def _concat_generators(self, *gen):
+    """Concatenates Python generators."""
+    for g in gen:
+      yield from g
+
+  def reshape(self, tensor: Tensor, shape: Tensor) -> Tensor:
+    tensor = tensor.reshape(shape)
+    return tensor
+
+  def transpose(self, tensor: Tensor, perm: Sequence[int]) -> Tensor:
+    shape = tuple(tensor.shape[i] for i in perm)
+    tensor = tensor.reshape(shape)
+    return tensor
+
+  def svd_decomposition(self,
+                        tensor: Tensor,
+                        split_axis: int,
+                        max_singular_values: Optional[int] = None,
+                        max_truncation_error: Optional[float] = None
+                       ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+    raise NotImplementedError("SVD shape cannot be calculated without"

```

Comment:
I think you should be able to implement this (You can leave it for a later PR), but everything except for max_truncation_error doesn't depend on the concrete tensor.